pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -107,40 +115,46 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioned, "")
+	return reconcile.Result{}, r.setStatusReady(userAcc)
 }
 
 func (r *ReconcileUserAccount) ensureUser(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (*userv1.User, bool, error) {
 	user := &userv1.User{}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user); err != nil {
 		if errors.IsNotFound(err) {
 			logger.Info("creating a new user", "name", userAcc.Name)
-			if err := r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, ""); err != nil {
+			if err := r.setStatusProvisioning(userAcc); err != nil {
 				return nil, false, err
 			}
 			user = newUser(userAcc)
 			if err := controllerutil.SetControllerReference(userAcc, user, r.scheme); err != nil {
-				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, err, "failed to set controller reference for user '%s'", userAcc.Name)
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusUserCreationFailed, err, "failed to set controller reference for user '%s'", userAcc.Name)
 			}
 			if err := r.client.Create(context.TODO(), user); err != nil {
-				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, err, "failed to create user '%s'", userAcc.Name)
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusUserCreationFailed, err, "failed to create user '%s'", userAcc.Name)
+			}
+			if err := r.setStatusUserCreationOK(userAcc); err != nil {
```

Comment:
I'm not really a fan of updating the status every time when either User, Identity or IdentityMapping is created - remember that every update causes an update of the MUR and status synchronization, so at least 5 times per one user provisioning which we don't really need (two are more than enough :-)). 
I think that we could modify the method so it sets the condition status only if the condition already exists - to cover the case when it would fail in a previous reconcile. WDYT?

Reply:
Well, it’s a question about optimization vs. more detailed status updates. I would prefer the later for now because of:
- user provisioning is not that frequent. So I don’t think that optimizations will give us much.
- in the beginning I think more details might be more useful than optimization. Later on we can revisit it.

Reply:
Ok

Reply:
On the other hand maybe just having Provisioning and Ready conditions would be enough... All the error messages with reasons can be in Ready condition.